core/src/main/java/feign/Response.java

```diff
@@ -105,6 +220,16 @@ public Body body() {
     return body;
   }
 
+  /**
+   * if present, the request that generated this response
```

Comment:
From a user POV, we guarantee this won't be null by the time they see it. I'd move the comment about null check to the Builder